torch/_inductor/lowering.py

```diff
@@ -1736,9 +1739,35 @@ def register_fallback(op_overload):
         )
 
     if isinstance(op, torch._ops.OpOverloadPacket):
-        for ol in op.overloads():
+        skip_overloads = []
+        overloads = []
+        for ol in op.overloads_with_kernels():
             op_overload = getattr(op, ol)
-            register_fallback(op_overload)
+            if op_overload in decompositions:
+                skip_overloads.append(op_overload)
+            else:
+                overloads.append(op_overload)
+        if overloads:
+            # Having some overloads in decompositions and others not,
+            # is very untypical. So, we'll be maximally verbose here.
+            for op_overload in skip_overloads:
+                log.warning(
+                    "Skipping to register fallback for %s that is in decompositions.",
+                    op_overload,
+                )
+            for op_overload in overloads:
+                if skip_overloads:
+                    log.warning(
+                        "Registering fallback for %s while there exists other overloads in decompositions.",
+                        op_overload,
+                    )
+                register_fallback(op_overload)
+        else:
+            log.warning(
```

Comment:
Could this be written as a test in CI, rather than running these checks at runtime?

Reply:
Yes I'm not concerned about performance but rather we can't raise an error if it happens at runtime. We could raise an error inside a test though, and that failing test will be more visible to reviewers.

Reply:
the plan is to turn these into hard errors. See the thread at https://github.com/pytorch/pytorch/pull/117640#discussion_r1455770402